pkg/migration-tool/planner/planner.go

```diff
@@ -0,0 +1,227 @@
+package planner
+
+import (
+	"context"
+	"fmt"
+	"go.uber.org/atomic"
+	"math"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/prometheus/pkg/labels"
+	"github.com/schollz/progressbar/v3"
+	"github.com/timescale/promscale/pkg/log"
+	"github.com/timescale/promscale/pkg/migration-tool/utils"
+)
+
+const (
+	MaxTimeRangeDeltaLimit    = time.Minute * 120
+	ResponseDataSizeHalfLimit = utils.Megabyte * 25
+)
+
+// Plan represents the plannings done by the planner.
+type Plan struct {
+	Mint                      int64
+	Maxt                      int64
+	JobName                   string
+	ProgressMetricName        string // Name for progress metric.
+	ProgressEnabled           bool
+	RemoteWriteStorageReadURL string
+	readClient                *utils.Client // Used to fetch last maxt pushed from write storage.
+	blockCounts               atomic.Int64  // Used in maintaining the ID of the in-memory blocks.
```

Comment:
I can't find this being used anywhere anymore

Reply:
It is actually used in the integration tests. Used to say how many blocks are sent by the read remote storage which actually means the number of samples in the progress metric.